### ISSUE TITLE
Modifying `fqdn` in an existing TunnelBinding does not clean up DNS records

### ISSUE THREAD

**Issue by Quantum64** [adyanth/cloudflare-operator#70]
DNS deletion is only executed in the finalizer.

**Comment by adyanth** [comment]
Yes, that is correct. I could not think of a way to do that since the operator does not receive the diff of what has been modified. 

We could check the current config and look at what changed, or mark the DNS entries with some unique operator ID and clean up everything when the binding is deleted, but I did not feel that was the most optimal solution. I will keep this issue open to track and find a proper solution. Do let me know if you have any ideas.

**Comment by Quantum64** [comment]
Given the flakey behavior and the ability to proxy star records, I would definitely prefer to disable Cloudflare API integration completely. The ability to programmatically manage the cloudflared configuration via CRDs is still very useful for us. Is there a timeline for #57?

**Comment by adyanth** [comment]
That is a good approach. I will work on that, should be done soon and straightforward to complete that as a conditional check before calling the APIs.

**Comment by adyanth** [comment]
@Quantum64 #71 is merged.
